frontend/src/pages/Compare.test.tsx

```diff
@@ -150,7 +150,166 @@ describe('Switch between v1 and v2 Run Comparison pages', () => {
+        'At least two runs and at most ten runs must be selected to view the Run Comparison page.',
+      message: 'Error: failed loading the Run Comparison page. Click Details for more information.',
+      mode: 'error',
+    });
+  });
+
+  it('Show no error on v1 page if run versions are mixed between v1 and v2 and feature flag disabled', async () => {
+    const getRunSpy = jest.spyOn(Apis.runServiceApi, 'getRun');
+    runs = [
+      newMockRun(MOCK_RUN_1_ID, false),
+      newMockRun(MOCK_RUN_2_ID, true),
+      newMockRun(MOCK_RUN_3_ID, true),
+    ];
+    getRunSpy.mockImplementation((id: string) => runs.find(r => r.run!.id === id));
+
+    // v2 feature is turn off.
+    jest.spyOn(features, 'isFeatureEnabled').mockImplementation(_ => false);
+
+    render(
+      <CommonTestWrapper>
+        <Compare {...generateProps()} />
+      </CommonTestWrapper>,
+    );
+
+    await TestUtils.flushPromises();
+
     await waitFor(() => screen.getByText('Run overview'));
+    expect(updateBannerSpy).toHaveBeenLastCalledWith({});
+  });
+
+  it('Show no error on v1 page if there are less than two runs and v2 feature flag disabled', async () => {
+    const getRunSpy = jest.spyOn(Apis.runServiceApi, 'getRun');
+    runs = [newMockRun(MOCK_RUN_1_ID, true)];
+    getRunSpy.mockImplementation((id: string) => runs.find(r => r.run!.id === id));
+
+    // v2 feature is turn off.
+    jest.spyOn(features, 'isFeatureEnabled').mockImplementation(_ => false);
+
+    const props = generateProps();
+    props.location.search = `?${QUERY_PARAMS.runlist}=${MOCK_RUN_1_ID}`;
+    render(
+      <CommonTestWrapper>
+        <Compare {...props} />
+      </CommonTestWrapper>,
+    );
+
+    await TestUtils.flushPromises();
+
+    await waitFor(() => screen.getByText('Run overview'));
+    expect(updateBannerSpy).toHaveBeenLastCalledWith({});
+  });
+
+  it('Show info banner on v1 page if all runs are v2 and v2 feature flag disabled', async () => {
+    const getRunSpy = jest.spyOn(Apis.runServiceApi, 'getRun');
+    runs = [
+      newMockRun(MOCK_RUN_1_ID, true),
+      newMockRun(MOCK_RUN_2_ID, true),
+      newMockRun(MOCK_RUN_3_ID, true),
+    ];
+    getRunSpy.mockImplementation((id: string) => runs.find(r => r.run!.id === id));
+
+    // v2 feature is turn off.
+    jest.spyOn(features, 'isFeatureEnabled').mockImplementation(_ => false);
+
+    render(
+      <CommonTestWrapper>
+        <Compare {...generateProps()} />
+      </CommonTestWrapper>,
+    );
+
+    await TestUtils.flushPromises();
+
+    await waitFor(() => screen.getByText('Run overview'));
+    expect(updateBannerSpy).toHaveBeenLastCalledWith({
+      additionalInfo:
+        'The selected runs are all V2, but the V2_ALPHA feature flag is disabled.' +
+        ' The V1 page will not show any useful information for these runs.',
+      message:
+        'Info: enable the V2_ALPHA feature flag in order to view the updated Run Comparison page.',
+      mode: 'info',
+    });
```

Comment:
Ok, that makes sense. I did not use `getByText` on these tests because the banner is not displayed on the `Compare.tsx` page but instead the `Router.tsx` page, so with the test in its existing form the `getByText` function call will not find the relevant text.

I could show the `Router` component via the `render` function, but it seems the [`expect(updateBannerSpy)`](https://github.com/kubeflow/pipelines/search?q=expect%28updateBannerSpy%29) approach was used on other tests so I decided to use that here as well. I'll note to prioritize using `getByText` in most circumstances though, and am happy to do it here as well if necessary.

Reply:
Right, the [expect(updateBannerSpy)](https://github.com/kubeflow/pipelines/search?q=expect%28updateBannerSpy%29) is a legacy convention which we hope we can move away from. It is okay to use updateBannerSpy when other approaches are not feasible. For now, you don't need to take action on the current PR code to switch to load `Router` component. It is just a note for future change.

Reply:
I didn't realize that was a legacy convention, that's good to know thank you. I'll be sure to note that for future change 👍